src/librustc_mir/shim.rs

```diff
@@ -196,7 +196,7 @@ fn build_drop_shim<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     let source_info = SourceInfo { span, scope: OUTERMOST_SOURCE_SCOPE };
 
     let return_block = BasicBlock::new(1);
-    let mut blocks = IndexVec::new();
+    let mut blocks = IndexVec::with_capacity(2);
```

Comment:
hmm do we want/need a `SmallIndexVec` that is analogous to `SmallVec` ...?

Reply:
or ... should we try something bonkers like changing `IndexVec` to *use* a `SmallVec` internally...?

Reply:
I could submit an experimental PR to check the perf implications.

Reply:
This is harder than I thought; changing the underlying structure of `IndexVec` to a `SmallVec` makes `Mir` a recursive type of infinite size due to the `promoted` component.